include/zephyr/devicetree.h

```diff
@@ -982,6 +984,67 @@
 	COND_CODE_1(DT_NODE_HAS_PROP(node_id, prop), \
 		(DT_STRING_UPPER_TOKEN(node_id, prop)), (default_value))
 
+/**
+ * @brief Get a string property's value as an unquoted string content.
+ *
+ * This removes "the quotes" from string-valued properties.
+ * That can be useful, for example,
+ * when defining floating point values as a string or a equation
+ * that may contain functions to call.
+ *
+ * DT_STRING_UNQUOTED() can only be used for properties with string type.
+ *
+ * It is an error to use DT_STRING_UNQUOTED() in other circumstances.
+ *
+ * Example devicetree fragment:
+ *
+ *     n1: node-1 {
+ *             prop = "12.7";
+ *     };
+ *     n2: node-2 {
+ *             prop = "0.5";
+ *     }
+ *     n3: node-3 {
```

Comment:
I'm timid; please remove this example :laughing:

Reply:
OK